### PR TITLE
My vote UI

### PR DESCRIPTION
**Pull request by Lt-Kartik-9PARA-SF**
My vote UI changes 
Vote history Modal implemented

### CODE REVIEW

File: src/components/Post/GovernanceSideBar/index.tsx [hunk 1]
```diff
@@ -694,6 +936,59 @@ const GovernanceSideBar: FC<IGovernanceSidebarProps> = (props) => {
...
+																	<div className=''>
+																		<p className='font-medium text-[12px] leading-6 text-[#243A57] mb-[5px]'>Last Vote:</p>
+																		<div className='flex justify-between text-[#243A57] text-[12px] font-normal leading-6 mb-[-5px]'>
+																			<Tooltip placement="bottom"  title="Decision"  color={'#E5007A'} className='max-w-[100px] max-[345px]:w-auto'>
+																				<span className='h-[25px]'>{vote.decision == 'yes' ? <p><AyeGreen /> <span className='capitalize font-medium text-[#2ED47A]'>{'Aye'}</span></p> :vote.decision == 'no' ?  <div><DislikeIcon className='text-[#F53C3C]'/> <span className='mb-[5px] capitalize font-medium text-[#F53C3C]'>{'Nay'}</span></div> : vote.decision == 'abstain' && (!(vote.balance as any).abstain)  ? <p><SplitYellow className='mb-[-2px]'/> <span className='capitalize font-medium text-[#FFBF60]'>{'Split'}</span></p>  : vote.decision == 'abstain' && (vote.balance as any).abstain ? <p className='flex justify-center align-middle'><AbstainGray className='mr-1 mb-[-8px]'/> <span className='capitalize font-medium  text-[#243A57]'>{'Abstain'}</span></p>: null }</span>
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/index.tsx:971]: Please extract this span to a seperate component

**Review by alphainfinitus — CHANGES_REQUESTED** [review]
Please check comments

File: src/components/Post/GovernanceSideBar/index.tsx (hunk 1 above)
Comment (Lt-Kartik-9PARA-SF) [reply src/components/Post/GovernanceSideBar/index.tsx:971]: will fix

File: src/components/Post/Post.tsx [hunk 2]
```diff
@@ -99,6 +100,7 @@ const Post: FC<IPostProps> = (props) => {
 
 	const isOnchainPost = checkIsOnChainPost(proposalType);
 	const isOffchainPost = !isOnchainPost;
+	const [lastVote, setLastVote] = useState< ILastVote | undefined >(undefined);
```
Comment (alphainfinitus) [inline src/components/Post/Post.tsx:103]: Why is the `lastVote` state stored here ? It's not being used here neither we're setting via `setLastVote` in this page.
It's only being passed to GovernanceSidebar so it should be in GovernanceSidebar

Comment (Lt-Kartik-9PARA-SF) [reply src/components/Post/Post.tsx:103]: when lastVote was in Governance side bar , it was causing an issue , state was not accessible when we were switching to mobile view from web view . to solve this issue moved lastVote state to the parent of Gov sidebar

File: src/components/Post/GovernanceSideBar/index.tsx [hunk 3]
```diff
@@ -159,7 +192,92 @@ const GovernanceSideBar: FC<IGovernanceSidebarProps> = (props) => {
 		supportThreshold: 0
 	});
 
+	const [vote,setVote] = useState<{
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/index.tsx:195]: `useState<ILastVoteInfoOnchainProps>`

File: src/components/Post/GovernanceSideBar/index.tsx [hunk 4]
```diff
@@ -62,6 +72,30 @@ interface IGovernanceSidebarProps {
 	tally?: any;
 	post: IPostResponse;
 	toggleEdit?: () => void;
+	lastVote: ILastVote | undefined;
+	setLastVote: React.Dispatch<React.SetStateAction<ILastVote | undefined>>
+}
+
+interface ILastVoteInfoOnchainProps{
+	isLastVoteLoading: boolean;
```
Comment (alphainfinitus) [inline src/components/Post/GovernanceSideBar/index.tsx:80]: `isLastVoteLoading` should be an optional field

**Comment by alphainfinitus** [comment]
Last Vote is shown for login address (or default address if no login addr. is available).

**Review by alphainfinitus — APPROVED** [review]

**Pull request merged**